politeiawww/util/proposal.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strings"
+
+	www "github.com/decred/politeia/politeiawww/api/v1"
+	"github.com/kennygrant/sanitize"
+)
+
+const (
+	// IndexFile contains the file name of the index file
+	IndexFile = "index.md"
+)
+
+var (
+	// ErrIndexFileNotFound indicates that the index file was not found in the
+	// proposal files.
+	ErrIndexFileNotFound = errors.New("index.md file was not found")
+)
+
+// ProposalName returns the proposal name
+func ProposalName(files []www.File) (string, error) {
+	var (
+		fileFound bool
+		payload   string
+		name      string
+	)
+
+	// get index file payload
+	for _, f := range files {
+		if f.Name == IndexFile {
+			payload = f.Payload
+			fileFound = true
+			break
+		}
+	}
+	if !fileFound {
+		return "", ErrIndexFileNotFound
```

Comment:
This error is due to bad user input, which means it should return:

    www.UserError{
        ErrorCode: ErrorStatusProposalMissingDescription,
    }

Errors other than `www.UserError` are reserved for unexpected things and get returned by the API as `500 Internal Server Errors`. Errors due to bad user input get returned (usually) as `400 Bad Request`.